secant/Features/App/App.swift

```diff
@@ -15,17 +15,42 @@ struct AppState: Equatable {
     var phraseValidationState: RecoveryPhraseValidationState
     var phraseDisplayState: RecoveryPhraseDisplayState
     var route: Route = .welcome
+    var storedWallet: StoredWallet?
+    var appInitialisationState: InitialisationState = .uninitialized
 }
 
 enum AppAction: Equatable {
-    case updateRoute(AppState.Route)
+    case appDelegate(AppDelegateAction)
+    case checkWalletInitialisation
+    case createNewWallet
     case home(HomeAction)
+    case initializeApp
     case onboarding(OnboardingAction)
     case phraseDisplay(RecoveryPhraseDisplayAction)
     case phraseValidation(RecoveryPhraseValidationAction)
+    case updateRoute(AppState.Route)
+    case walletInitialisationResult(InitialisationState)
+}
+
+struct AppEnvironment {
+    let mainQueue: AnySchedulerOf<DispatchQueue>
```

Comment:
I think you'll want to name this one scheduler.  Not only is this a type erased scheduler, it's also passed to functions that ask for a scheduler.  The fact that you end up using the main queue is just an implementation detail.

Reply:
fixed